### PR TITLE
[WIP] CI pipeline test, no merging intended

### PR DESCRIPTION
**Pull request by maxaehle**
**This PR was created to trigger the CI pipeline, no merging intended.**

I noticed differences between results of regression tests, when running them on my system and by the Github CI pipeline. Let me check whether according to the CI pipeline, the wall roughness was actually always 0 before my bugfix in #1286.

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [ ] I am submitting my contribution to the develop branch.
- [ ] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags, or simply --warnlevel=2 when using meson).
- [ ] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

File: Common/src/geometry/CGeometry.cpp [hunk 1]
```diff
@@ -3942,5 +3942,36 @@ void CGeometry::ComputeWallDistance(const CConfig* const* config_container, CGeo
...
+        });
+      NdFlattener<2> roughness_local(roughness_f);
+      NdFlattener<2> const& roughness_local_c = roughness_local;
+      std::cout << "roughness_local[0][0]=" << roughness_local[0][0] << "\n";
+      std::cout << "const roughness_local[0][0]=" << roughness_local_c[0][0] << "\n";
+      // [rank][iZone][iMarker] -> roughness
+      NdFlattener<3> roughness_global(Nd_MPI_Environment(), roughness_local);
+      std::cout << "roughness_global[0][0][0]=" << roughness_global[0][0][0] << "\n";
+      NdFlattener<3> const& roughness_global_c = roughness_global;
+      std::cout << "const roughness_global[0][0][0]=" << roughness_global_c[0][0][0] << "\n";
```
Comment (maxaehle) [inline Common/src/geometry/CGeometry.cpp:3968]: Output in the serial AD testcase discadj_incomp_cylinder:

    ...
    Computing wall distances.
    roughness_local[0][0]=0
    const roughness_local[0][0]=0
    roughness_global[0][0][0]=-nan
    const roughness_global[0][0][0]=-nan
    GlobalIndex.size()=14576
    SetWallRoughness 1: set iPoint=0 roughness=-nan
    ...

File: Common/include/toolboxes/ndflattener.hpp [hunk 2]
```diff
@@ -0,0 +1,860 @@
...
+    std::cout << "parameter to MPI_Allgatherv set_g K="<<K<<" were: "
+             << local_version.indices.data()+1 << ";"
+             << Nodes_all[K - 1][mpi_env.rank] << ";"
+             <<mpi_env.mpi_index << ";"
+             <<indices.data()<<";"
+             <<Nodes_all_K_as_int.data()<<";"
+             <<Nodes_all_k_cumulated.data()<<";"
+             <<mpi_env.mpi_index<<";"
+             << mpi_env.comm<<"\n";
+    std::cout << "here, Nodes_all_K_as_int[0]="<<Nodes_all_K_as_int[0]<<", Nodes_all_k_cumulated[0]="<<Nodes_all_k_cumulated[0]<<" size="<<mpi_env.size<<"\n";
```
Comment (maxaehle) [inline Common/include/toolboxes/ndflattener.hpp:690]: CI output for serial AD discadj_incomp_cylinder:

    indices[0] before Allgatherv: 0
    indices[0] after Allgatherv: 2
    parameter to MPI_Allgatherv set_g K=2 were: 0x560568495898;1;1;0x560568846750;0x56056889d270;0x560568ba2210;1;0
    here, Nodes_all_K_as_int[0]=1, Nodes_all_k_cumulated[0]=1 size=1

indices[0] should remain 0...

**Comment by maxaehle** [comment]
In `mpi_structure.hpp`:

    static inline void Allgatherv(const void* sendbuf, int sendcnt, Datatype sendtype, void* recvbuf, const int* recvcnt, const int* displs, Datatype recvtype, Comm comm) {
      CopyData(sendbuf, recvbuf, sendcnt, sendtype);
    }

ignores `displs`, this looks fishy

**Comment by pcarruscag** [comment]
Maybe it needs `CopyData(sendbuf, recvbuf+displs[0], sendcnt, sendtype);` ?

**Comment by maxaehle** [comment]
Yes, after this kind of correction the tests succeed, here and in #1286! (Shifts are handed to `CopyData` and added there, because `recvbuf` is a void pointer). Closing this PR now.

**Pull request closed without merging**